atintegrators/atimplib.c

```diff
@@ -306,15 +304,16 @@ void compute_kicks_longres(int nslice,int nbunch,int nturns, double *turnhistory
 
     int rank=0;
     int size=1;
-    int i,ii,ib;
-    double ds,wi;
+    int i,ii,ib,loopstart,loopend;
+    double ds,wi,wii;
     double *turnhistoryZ = turnhistory+nslice*nbunch*nturns*2;
     double *turnhistoryW = turnhistory+nslice*nbunch*nturns*3;
-    double *wake = malloc(2*sizeof(double));
+    double *wake = atMalloc(2*sizeof(double));
```

Comment:
Why are you allocating dynamically an array of fixed length ?
```C
double wake[2];
```
is more efficient and does not require freeing.